src/parsing/preparse-data.cc

```diff
@@ -307,13 +307,7 @@ bool PreparseDataBuilder::SaveDataForSkippableFunction(
   return has_data;
 }
 
-void PreparseDataBuilder::SaveScopeAllocationData(DeclarationScope* scope,
-                                                  Parser* parser) {
-  if (!has_data_) return;
-  DCHECK(HasInnerFunctions());
-
-  byte_data_.Start(parser->preparse_data_buffer());
-
+void PreparseDataBuilder::SaveScopeAllocationDataImpl(DeclarationScope* scope) {
```

Comment:
had to pull this block of SaveScopeAllocationData out from the header since DeclarationScope was an incomplete type there. this was my best take at dealing with that but if there's a better approach lmk!

Reply:
This is probably fine. If the performance ends up mattering (which it doesn't seem like will be the case for this method), we can always create a secondary header specifically for inline functions.